src/sentry/release_health/metrics_sessions_v2.py

```diff
@@ -344,26 +400,32 @@ def _get_snuba_query(
             # exclude unresolved keys from groupby
             pass
 
-    full_groupby = set(groupby.values())
+    full_groupby = list(set(groupby.values()))
 
     if series:
-        full_groupby.add(Column(TS_COL_GROUP))
+        full_groupby.append(Column(TS_COL_GROUP))
 
     query_args = dict(
         dataset=Dataset.Metrics.value,
         match=Entity(entity_key.value),
-        select=list(columns),
-        groupby=list(full_groupby),
+        select=columns,
+        groupby=full_groupby,
         where=conditions,
         granularity=Granularity(query.rollup),
     )
 
-    if series:
-        # Set higher limit and order by to be consistent with sessions_v2
-        query_args["limit"] = Limit(SNUBA_LIMIT)
-        query_args["orderby"] = [OrderBy(Column(TS_COL_GROUP), Direction.DESC)]
+    # In case of group by, either set a limit or use the groups from the
+    # first query to limit the results:
+    if query.raw_groupby:
+        if limit_state.limiting_conditions is None:
+            # Set limit and order by to be consistent with sessions_v2
+            max_groups = SNUBA_LIMIT // len(get_timestamps(query))
+            query_args["limit"] = Limit(max_groups)
+            query_args["orderby"] = [OrderBy(columns[0], Direction.DESC)]
+        else:
+            query_args["where"] += limit_state.limiting_conditions
 
-    return Query(**query_args)
+    return Query(**query_args), full_groupby
```

Comment:
Why do you need to return `full_groupby` here if its practically part of the `query_args` and you are returning `Query(**query_args)`, can't you just access that on the query later object or even just return the `query_args` and instantiate the `Query` object later?

Reply:
Good point, changed to return only `Query`.